fem/pgridfunc.cpp

```diff
@@ -908,6 +908,100 @@ void ParGridFunction::SaveAsOne(const char *fname, int precision) const
    SaveAsOne(ofs);
 }
 
+void ParGridFunction::SaveAsSerial(const char *fname, int precision,
+                                   int save_rank) const
+{
+   GridFunction serialgf = GetSerialGridFunction(save_rank);
+   ParMesh *pmesh = ParFESpace()->GetParMesh();
+   if (pmesh->GetMyRank() == save_rank)
+   {
+      serialgf.Save(fname, precision);
+   }
+   MPI_Barrier(pmesh->GetComm());
+}
+
+GridFunction ParGridFunction::GetSerialGridFunction(int save_rank) const
+{
+   ParFiniteElementSpace *pfespace = ParFESpace();
+   ParMesh *pmesh = pfespace->GetParMesh();
+
+   Mesh serialmesh = pmesh->GetSerialMesh(save_rank);
+
+   int vdim = pfespace->GetVDim();
+   auto *fespace_serial = new FiniteElementSpace(&serialmesh,
```

Comment:
Does this cause a memory leak?  If so, I'd ask for either:
- addition of `delete fespace_serial;` somewhere
- use of smart pointer, i.e., `std::unique_ptr<FiniteElementSpace> fespace_serial(new FiniteElementSpace(...));`

Reply:
good catch @cjvogl

Reply:
I fixed the unit test in latest commit by having the `GridFunction` own the `FiniteElementSpace` and `FECollection`.. this makes sure the `FiniteElementSpace` is deleted when the `GridFunction` is deleted..